### PR TITLE
[tests-only][full-ci]tests: add test for lock/unlock in public link share

### PR DESCRIPTION
**Pull request by SwikritiT**
## Related Issue

-  part of: https://github.com/owncloud/ocis/issues/7625

### CODE REVIEW

File: tests/acceptance/features/apiLocks/unlockFiles.feature [hunk 1]
```diff
@@ -274,3 +274,27 @@ Feature: unlock locked items
...
+    Given using <dav-path-version> DAV path
+    And user "Alice" has created folder "PARENT"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "PARENT/textfile0.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | PARENT   |
+      | permissions | change   |
+      | password    | %public% |
+    And the public has locked "textfile0.txt" in the last public link shared folder setting the following properties
+      | lockscope | <lock-scope> |
+   When the public unlocks file "textfile0.txt" using the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiLocks/unlockFiles.feature:289]:
```suggestion
    When the public unlocks file "textfile0.txt" using the WebDAV API
```

**Review by saw-jan — APPROVED** [review]

File: tests/acceptance/features/apiLocks/lockFiles.feature [hunk 2]
```diff
@@ -386,3 +409,93 @@ Feature: lock files
...
+    And 0 locks should be reported for file "PARENT/textfile0.txt" of user "Alice" by the WebDAV API
+    And user "Alice" should be able to upload file "filesForUpload/lorem.txt" to "PARENT/textfile0.txt"
+    Examples:
+      | dav-path-version | lock-scope |
+      | old              | shared     |
+      | old              | exclusive  |
+      | new              | shared     |
+      | new              | exclusive  |
+      | spaces           | shared     |
+      | spaces           | exclusive  |
```
Comment (SagarGi) [inline tests/acceptance/features/apiLocks/lockFiles.feature:457]: can this scenario be combined into 1?

File: tests/acceptance/features/apiLocks/lockFiles.feature [hunk 3]
```diff
@@ -386,3 +409,93 @@ Feature: lock files
...
+    And 0 locks should be reported for file "textfile0.txt" of user "Alice" by the WebDAV API
+    And user "Alice" should be able to upload file "filesForUpload/lorem.txt" to "textfile0.txt"
+    Examples:
+      | dav-path-version | lock-scope |
+      | old              | shared     |
+      | old              | exclusive  |
+      | new              | shared     |
+      | new              | exclusive  |
+      | spaces           | shared     |
+      | spaces           | exclusive  |
```
Comment (SagarGi) [inline tests/acceptance/features/apiLocks/lockFiles.feature:501]: Also this one as well?

File: tests/acceptance/features/apiLocks/lockFiles.feature (hunk 2 above)
Comment (SwikritiT) [reply tests/acceptance/features/apiLocks/lockFiles.feature:457]: maybe I'll check

Comment (SwikritiT) [reply tests/acceptance/features/apiLocks/lockFiles.feature:457]: @SagarGi No the scenario cannot be combined into one. One scenario is the case where the anon user has only `read ` permission so they should not be able to lock the file and another is where they have `edit` permission where they should be able to lock the file

File: tests/acceptance/features/apiLocks/lockFiles.feature (hunk 3 above)
Comment (SwikritiT) [reply tests/acceptance/features/apiLocks/lockFiles.feature:501]: same as above

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

**Pull request merged**